app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/HomeFragment.java

```diff
@@ -224,7 +224,7 @@ public void onCustomTabsDisconnected() {
 
     private void getTagline() {
         OpenFoodAPIService openFoodAPIService = new OpenFoodAPIClient(getActivity(), "https://ssl-api.openfoodfacts.org").getAPIService();
-        Call<ArrayList<TaglineLanguageModel>> call = openFoodAPIService.getTagline();
+        Call<ArrayList<TaglineLanguageModel>> call = openFoodAPIService.getTagline(getString(R.string.app_name) + " " + Utils.getUserAgent("OTHERS"));
```

Comment:
samething here. The prefix should be added by `Utils.getUserAgent`

In method `Utils.getUserAgent`, we could use `BuildConfig.APPLICATION_ID` as prefix and not `Official Android App `

Reply:
@teolemon  is it for you ? I don't know the impact for the analytics part ?
The user agent will be:
openfoodfacts.github.scrachx.openbeauty 3.2.2 THE_TYPE

Reply:
There is something wrong in the method `getUserAgent`.
The parameter `type` is not totally used ( only to test HEADER_USER_AGENT_SCAN / HEADER_USER_AGENT_SEARCH)

I suggest to replace: 
> public static String getUserAgent(String type) {
        final String prefix = "Official Android App ";
        if (type.equals(HEADER_USER_AGENT_SCAN)) {
            return prefix + BuildConfig.VERSION_NAME + " " + HEADER_USER_AGENT_SCAN;
        } else if (type.equals(HEADER_USER_AGENT_SEARCH)) {
            return prefix + BuildConfig.VERSION_NAME + " " + HEADER_USER_AGENT_SEARCH;
        }
        return prefix + BuildConfig.VERSION_NAME;
    }


by 

> public static String getUserAgent(String type) {
>         return BuildConfig.APPLICATION_ID+" " + BuildConfig.VERSION_NAME+" "+type;
> }

@huzaifaiftikhar please could you change this method

Reply:
@deniger I will make the changes suggested by you but will also overload this method without an argument so that we can get the user agent without THE_TYPE if required (which is required in the current scenario). 
Also, instead of using `BuildConfig.APPLICATION_ID` I will add another BuildConfig variable with the APP_NAME. This would look much better than the application ID. So we would be having the same user agent everywhere in the code.